content/docs/reference/buildpack-api.md

```diff
@@ -0,0 +1,142 @@
+the buildpack will execute against.
+
+The executable must return an exit code of `0` if the codebase can be serviced by this buildpack. If the exit code is `0`, the script must print a human-readable name to `stdout`.
+
+### Example
+
+This is a simple example of a buildpack that detects a Python application by
+checking for the presence of a `requirements.txt` file:
+
+```
+#!/bin/sh
+
+if [ -f requirements.txt ]; then
+  echo "Python"
+  exit 0
+else
+  exit 1
+fi
+```
+
+## `bin/build`
+
+### Usage
+
+```
+bin/build LAYERS_DIR PLATFORM_DIR BUILD_PLAN
+```
+
+This entrypoint transforms a codebase.
+It will often resolve dependencies, install binary packages, and compile code.
+It accepts three positional arguments:
+
+* `LAYERS_DIR` - a directory that may contain subdirectories representing each layer created by the buildpack in the final image or build cache.
+* `PLATFORM_DIR` - a directory containing platform provided configuration, such as environment variables.
+* `BUILD_PLAN` - a string containing the [Build Plan](https://github.com/buildpack/spec/blob/master/buildpack.md#buildpack-plan-toml).
+
+In addition, the working directory is defined as the location of the codebase
+this buildpack will execute against.
+
+All changes to the codebase in the working directory will be included in the final
+image, along with any launch layers created in the `LAYERS_DIR`.
+
+#### Layers
+
+Each directory created by the buildpack under the `LAYERS_DIR` can be used for any
+of the following purposes:
+
+* Launch - the directory will be included in the run image as a single layer
+* Cache - the directory will be included in the cache
+* Build - the directory will be accessible by subsequent buildpacks
+
+A buildpack defines how a layer will by used by creating a `<layer>.toml` with
+a name matching the directory it describes in the `LAYERS_DIR`. For example, a
+buildpack might create a `$LAYERS_DIR/python` directory and a `$LAYERS_DIR/python.toml`
+with the following contents:
+
+```
+launch = true
+cache = true
+build = true
+```
+
+In this example, the `python` directory will be included in the run image,
+cached for future builds, and will be accessible to subsequent buildpacks.
+
+### Example
+
+This is a simple example of a buildpack that runs Python's `pip` package manager
+to resolve dependencies:
+
+```
+#!/bin/sh
+
+LAYERS_DIR="$1"
+
+PIP_LAYER="$LAYERS_DIR/pip"
+mkdir -p "$PIP_LAYER/modules" "$PIP_LAYER/env"
+
+pip install -r requirements.txt -t "$PIP_LAYER/modules" \
+  --install-option="--install-scripts=$PIP_LAYER/bin" \
+  --exists-action=w --disable-pip-version-check --no-cache-dir
+
+echo "launch = true" > $PIP_LAYER.toml
+```
+
+## `buildpack.toml`
+
+A buildpack must contain a `buildpack.toml` file in its root directory. This
+descriptor file must contain a `buildpack.id` that is a globally unique
+identifier.
+
+### Example
+
+```
+[buildpack]
+id = "example.com/python"
+version = "1.0"
+```
```

Comment:
`Pack` has validation that makes it a requirement to have EITHER `stacks` or `order`. I don't see that restriction explicitly in the spec but instead...

> If an order is specified, then stacks MUST NOT be specified.

https://github.com/buildpack/spec/blob/master/buildpack.md#buildpacktoml-toml

We probably need to align something here.